kernel/init.c

```diff
@@ -410,8 +405,11 @@ static inline void _initialize_stack_canaries(void)
 	 * devices are available should be built, this is only a fallback for
 	 * those devices without a HWRNG entropy driver.
 	 */
-	__stack_chk_guard = (uintptr_t)sys_rand32_get();
+	return sys_rand32_get();
 }
+
+#ifdef CONFIG_STACK_CANARIES
+extern uintptr_t __stack_chk_guard;
```

Comment:
Seems like this headerless extern has been passed around enough.  Move to kernel_structs.h maybe?

Reply:
it's defined by the compiler though, would we want to put it there?